### ISSUE TITLE
Require non-null mainStats for EmailDiagramHoverCard

### ISSUE THREAD

**Issue by WULCAN** [zetkin/app.zetkin.org#2257]
## Description

A developer should be able to maintain `EmailDiagramHoverCard` without considering null `mainStats`.

`EmailDiagramHoverCard` accepts null for `mainStats`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/components/EmailDiagramHoverCard.tsx#L14

The only use is in `OpenedInsightsSection` where `stats.data` is passed as `mainStats`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/components/OpenedInsightsSection.tsx#L238-L240

`stats` comes from `useEmailStats`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/components/OpenedInsightsSection.tsx#L77

where any `data` must come from `futureToObject`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/hooks/useEmailStats.ts#L76-L91

which copies `data` from the future:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/core/caching/futures.ts#L9-L11

`statsFuture` is either the result of `loadItemIfNecessary`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/hooks/useEmailStats.ts#L37

or `statsFuture` is a `ResolvedFuture` with `null` data if `useEmail` returned an object with false `isTargeted`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/hooks/useEmailStats.ts#L48-L50

or `statsFuture` is a `PlaceholderFuture` with non-null `data`.

`loadItemIfNecessary` returns either a `PromiseFuture` with initial `data` from `remoteItem`, potentially later resolving to the future's data, or `loadItemIfNecessary` returns a `RemoteItemFuture` with `data` from `remoteItem`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/core/caching/cacheUtils.ts#L85-L95

`remoteItem` comes from `useAppSelector` which is `useSelector` of `react-redux`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/hooks/useEmailStats.ts#L35

`statsItem`is either `undefined`, or `remoteItem` with non-null `data`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/store.ts#L139-L140
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/store.ts#L162-L165

If  `statsItem` is `undefined`, `loadItemIfNecessary` will return a `PromiseFuture` with initial `null` data.

`OpenedInsightsSection` only renders if `email.published` which shouldn't be possible if `isTargeted` is false:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/components/OpenedInsightsSection.tsx#L86-L88

While a `PromiseFuture` is pending, its `isLoading` is true and if it rejects, its `error` will likely be non-null. If it were still `isLoading`, `useEmailStats` would have replaced `statsFuture` with a `PlaceholderFuture` with non-null `data`:
https://github.com/zetkin/app.zetkin.org/blob/c279be9988118c47fe8d1db5e6339914aefb16d2/src/features/emails/hooks/useEmailStats.ts#L52-L53

In conclusion, the only way `EmailDiagramHoverCard` can get sent a null `mainStats`, is if the loading of stats fails. With a little error handling in `OpenedInsightsSection`, it seems possible to express in Typescript that `mainStats` cannot be null.

## Estimated size (S, M, L)

I think this issue is medium because a solution could require quite some refactoring. I don't think its large as the problem is relatively confined to email insights. The error handling might require some design.
